pkg/nuctl/test/project_test.go

```diff
@@ -200,24 +207,30 @@ func (suite *projectExportImportTestSuite) TestImportProject() {
 	uniqueProjectConfigPath := suite.addUniqueSuffixToImportConfig(projectConfigPath,
 		uniqueSuffix,
 		[]string{function1Name, function2Name},
-		[]string{function1EventDisplayName, function2EventDisplayName})
+		[]string{function1EventDisplayName, function2EventDisplayName},
+		nil)
 	defer os.Remove(uniqueProjectConfigPath) // nolint: errcheck
 
 	function1Name = function1Name + uniqueSuffix
 	function2Name = function2Name + uniqueSuffix
 	function1EventDisplayName = function1EventDisplayName + uniqueSuffix
 	function2EventDisplayName = function2EventDisplayName + uniqueSuffix
 
-	defer suite.ExecuteNuctl([]string{"delete", "proj", projectName}, nil) // nolint: errcheck
+	suite.createProject(projectName)
+	suite.createFunction(function1Name, projectName)
+
 	defer suite.ExecuteNuctl([]string{"delete", "fu", function1Name}, nil) // nolint: errcheck
 	defer suite.ExecuteNuctl([]string{"delete", "fu", function2Name}, nil) // nolint: errcheck
+	defer suite.ExecuteNuctl([]string{"delete", "proj", projectName}, nil) // nolint: errcheck
```

Comment:
Do the defers first, or each defer after the relevant creation.
Here, if createFunction fails, the project won't be removed because the defer is not scheduled